projects/opendr_ws_2/src/opendr_perception/opendr_perception/object_detection_2d_gem_node.py

```diff
@@ -0,0 +1,281 @@
+                    [213, 363],
+                    [390, 364],
+                    [212, 368],
+                    [390, 370],
+                    [467, 324],
+                    [415, 364],
+                ]
+            )
+            self.get_logger().warn(
+                "\nUsing default calibration values for pts_infra!"
+                + "\nThese are probably incorrect."
+                + "\nThe correct values for pts_infra can be found by running get_color_infra_alignment.py."
+                + "\nThis file is located in the opendr/perception/object_detection2d/utils module."
+            )
+        if pts_color is None:
+            pts_color = np.array(
+                [
+                    [910, 397],
+                    [889, 572],
+                    [874, 552],
+                    [891, 411],
+                    [635, 385],
+                    [619, 525],
+                    [603, 544],
+                    [576, 682],
+                    [810, 619],
+                    [216, 688],
+                    [90, 423],
+                    [281, 310],
+                    [193, 163],
+                    [684, 449],
+                    [806, 431],
+                    [504, 170],
+                    [24, 538],
+                    [353, 552],
+                    [323, 550],
+                    [323, 529],
+                    [344, 387],
+                    [961, 533],
+                    [570, 233],
+                    [392, 336],
+                    [831, 610],
+                    [378, 638],
+                    [742, 630],
+                    [378, 648],
+                    [742, 640],
+                    [895, 550],
+                    [787, 630],
+                ]
+            )
+            self.get_logger().warn(
+                "\nUsing default calibration values for pts_color!"
+                + "\nThese are probably incorrect."
+                + "\nThe correct values for pts_color can be found by running get_color_infra_alignment.py."
+                + "\nThis file is located in the opendr/perception/object_detection2d/utils module."
+            )
+        # Object classes
+        self.classes = ["N/A", "chair", "cycle", "bin", "laptop", "drill", "rocker"]
+
+        # Estimating Homography matrix for aligning infra with RGB
+        self.h, status = cv2.findHomography(pts_infra, pts_color)
+
+        self.bridge = ROS2Bridge()
+
+        # Initialize the detection estimation
+        model_backbone = "resnet50"
+
+        self.gem_learner = GemLearner(
+            backbone=model_backbone,
+            num_classes=7,
+            device=device,
+        )
+        self.gem_learner.fusion_method = "sc_avg"
+        self.gem_learner.download(path=".", verbose=True)
+
+        # Subscribers
+        msg_rgb = message_filters.Subscriber(self, ROS_Image, input_rgb_image_topic)
+        msg_ir = message_filters.Subscriber(self, ROS_Image, input_infra_image_topic)
```

Comment:
Similar to ROS1 node, both subscribers should have their queues set to 1 to avoid delays (issue #275).
```suggestion
        msg_rgb = message_filters.Subscriber(self, ROS_Image, input_rgb_image_topic, 1)
        msg_ir = message_filters.Subscriber(self, ROS_Image, input_infra_image_topic, 1)
```